torch/testing/_internal/common_device_type.py

```diff
@@ -788,6 +788,72 @@ def test_wrapper(*args, **kwargs):
                     print("Failed to instantiate {0} for op {1}!".format(test_name, op.name))
                     raise ex
 
+# Decorator that generates two set of tests with layout=torch.strided and layout=specified_layout.
+# All keyword arguments are used only for the specified layout, so they have no effect on the original test.
+class enableLayout(_TestParametrizer):
+    def __init__(self, layout: torch.layout, *,
+                 dtypes: Optional[Sequence[torch.dtype]] = None,
+                 dtypes_cpu: Optional[Sequence[torch.dtype]] = None,
+                 dtypes_cuda: Optional[Sequence[torch.dtype]] = None,
+                 only_cuda: bool = False,
+                 skip_cuda_if_no_cusparse: bool = True):
+        if not isinstance(layout, torch.layout):
+            raise RuntimeError("Expected to get torch.layout as layout, but got ", layout)
```

Comment:
NIt: 

1. Since we are checking the type here, it would make sense to also report the type back in the message
2. f-strings :stuck_out_tongue: 

```suggestion
            raise RuntimeError(f"Expected to get torch.layout as layout, but got {type(layout)} instead.")
```